lhotse/features/opensmile.py

```diff
@@ -0,0 +1,180 @@
+            sampling_rate=self.config.sampling_rate,
+            options=self.config.options,
+            loglevel=self.config.loglevel,
+            logfile=self.config.logfile,
+            channels=self.config.channels,
+            mixdown=self.config.mixdown,
+            resample=self.config.resample,
+            num_workers=self.config.num_workers,
+            verbose=self.config.verbose,
+        )
+
+    @property
+    def feature_names(self) -> List[str]:
+        return self.smileExtractor.feature_names
+
+    def is_lld_or_lld_de(self) -> bool:
+        from opensmile import FeatureLevel
+
+        return (
+            self.feature_level is FeatureLevel.LowLevelDescriptors
+            or self.feature_level is FeatureLevel.LowLevelDescriptors_Deltas
+        )
+
+    @property
+    def frame_shift(self) -> Seconds:
+        import opensmile
+
+        if (
+            self.is_lld_or_lld_de()
+            and self.feature_set in opensmile.FeatureSet.__members__.values()
+        ):
+            # For all deafult opensmile configs frameshift is equal to 10 ms
+            return 0.01
+        else:
+            raise NotImplementedError(
+                f"frame_shift is not defined for Functionals feature level or for non default feature set. Defined featureset: {self.config.feature_set}"
+            )
+
+    def feature_dim(self, sampling_rate: int) -> int:
+        return len(self.feature_names)
+
+    def feature_names(self) -> List[str]:
+        return self.smileExtractor.feature_names()
+
+    def extract(self, samples: np.ndarray, sampling_rate: int) -> np.ndarray:
+        if (
+            self.config.sampling_rate is not None
+            and self.config.sampling_rate != sampling_rate
+        ):
+            raise ValueError(
+                f"Given sampling rate ({sampling_rate}) mismatched with the value set in OpenSmileConfig ({self.config.sampling_rate})."
+            )
+        import opensmile
+
+        feats = self.smileExtractor.process_signal(
+            samples, sampling_rate=sampling_rate
+        ).to_numpy()
+
+        if self.is_lld_or_lld_de():
+            feats = self._pad_frames(samples, feats, sampling_rate)
+
+        return feats.copy()
+
+    def _pad_frames(
+        self, samples: np.ndarray, feats: np.ndarray, sampling_rate: int
+    ) -> np.ndarray:
+        """Adds last diff frames to the end of feats matrix to fit lhotse.utils.compute_num_frames."""
+        duration = np.shape(samples)[1] / sampling_rate
+        diff = (
+            compute_num_frames(duration, self.frame_shift, sampling_rate)
+            - np.shape(feats)[0]
+        )
+        if abs(diff) >= 6:
+            warnings.warn(f"Unusual difference in number of frames: {diff}")
+        if diff > 0:
+            feats = np.append(feats, feats[-diff:, :], axis=0)
+        elif diff < 0:
+            feats = feats[:-diff, :]
+        return feats
+
+
+#    @staticmethod
+#    def mix(
```

Comment:
Shall we remove such blocks?
If we need it later, we can always find it in the git history.